mcs/class/corlib/Test/System.Reflection/MethodInfoTest.cs

```diff
@@ -994,6 +994,16 @@ public void TestLocalVariableTypes ()
 				Assert.IsTrue (foundExpectedType, "#2-4");
 		}
 #endif
+
+		delegate int D1 ();
+
+		[Test]
+		public void Issue10838 ()
+		{
+			var flags = BindingFlags.DeclaredOnly | BindingFlags.Public | BindingFlags.Instance;
+			var members = typeof (D1).GetMembers (flags).OrderBy (m => m.MetadataToken);
```

Comment:
Can we just get the '.ctor' method directly? Why should this test depend on the order of methods (and their tokens) in a class?

Reply:
It's from original repro sample.

Reply:
Tweak the test, order by MetadataToken is not reliable